File: Java-SpringBoot/BE-01-InMemory/java-17-springboot-3.5/src/main/java/com/flyrank/userservice/config/OpenAPIConfig.java
package com.flyrank.userservice.config;

import io.swagger.v3.oas.models.OpenAPI;
import io.swagger.v3.oas.models.info.Info;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;

/**
 * OpenAPI metadata configuration for Swagger UI docs.
 */
@Configuration
public class OpenAPIConfig {

    @Bean
    public OpenAPI customOpenAPI() {
        return new OpenAPI()
                .info(new Info()
                        .title("Flyrank BE-01 User Service (Java 17 & Spring Boot 3.5)")
                        .version("1.0.0")
                        .description("A Spring Boot backend that creates and retrieves users, "
                                + "keeping all data in memory. Java 17 re-implementation of Flyrank assignment BE-01."));
    }
}

File: Java-SpringBoot/BE-01-InMemory/kotlin-2.3-springboot-3.5/src/main/kotlin/com/flyrank/userservice/config/OpenAPIConfig.kt
package com.flyrank.userservice.config

import io.swagger.v3.oas.models.OpenAPI
import io.swagger.v3.oas.models.info.Info
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration

@Configuration
class OpenAPIConfig {

    @Bean
    fun customOpenAPI(): OpenAPI {
        return OpenAPI()
            .info(
                Info()
                    .title("Flyrank BE-01 User Service (Kotlin 2.3 & K2 Compiler)")
                    .version("1.0.0")
                    .description("A Spring Boot 3.5 backend written in idiomatic Kotlin with K2 compiler, Data Classes, and Coroutines.")
            )
    }
}
